ceph/ceph_admin/helper.py

```diff
@@ -831,6 +832,57 @@ def validate_log_file_after_enable(cls):
     return True
 
 
+def validate_log_rotate(cls):
+    """
+    Verify logrotation of cephadm.log file
+    Args:
+        cls: cephadm instance object
+
+    Returns:
+        boolean
+    """
+    # step 1: check for the files in current log directory
+    out = cls.installer.exec_command(cmd="ls /var/log/ceph", sudo=True)
+    files = re.split("\n| ", out[0])
+    logFileExists = [f for f in files if f == "cephadm.log"]
+    if not logFileExists:
+        LOG.error("Log rotation failed: Logs not present in /var/log/ceph folder")
+        return False
+
+    # step 2: apply logrotate on cephadm.log
+    out = cls.installer.exec_command(
+        cmd="logrotate --force /etc/logrotate.d/cephadm", sudo=True
+    )
+    # if logrotate doesn't result in logs being zipped return False
+    out = cls.installer.exec_command(cmd="ls /var/log/ceph", sudo=True)
```

Comment:
I would suggest to add to check time for file as well. It should be after log rotate command execution.

Reply:
This would not be required, since cephadm.log would be converted into a gz file when we execute logrotate command. And the log file wouldn't exist. After an operation which generates new logs a new cephadm.log file would be created. So I feel its okay to not have a time check on the file.

Reply:
Then gz file might have name with timestamp. Add step to validate gz file.

Reply:
Done.